### ISSUE TITLE
Receiver reading message from AMQP:EOF error

### ISSUE THREAD

**Issue by sohamsengupta17** [Azure/go-amqp#64]
When running the example code included in the README.md, the receiver throws the following error approximately 1 min after receiving the last message from the broker queue: 
```
2021/09/15 16:15:49 Reading message from AMQP:EOF
exit status 1 
```

The issue occurs while executing the following line of the example:
```
for {
	// Receive next message
	msg, err := receiver.Receive(ctx)
	if err != nil {
		log.Fatal("Reading message from AMQP:", err)
	}

```
Since it is executing in an infinite for-loop, it was expected to continue receiving messages until explicitly closed. 
Using the HandleMessage method in place of Receive did not help. I have also tried changing the ConnIdleTimeout to both zero and non-zero values, and using a plain Background context without timeouts, but it did not have any effect. 

I am using Solace PubSub+ event broker for the message queue. I have not used any SASL authentication.

Any help would be greatly appreciated.

**Comment by niteshagrawalgmail** [comment]
👍

**Comment by jhendrixMSFT** [comment]
I wonder if this is due to our non-compliant idle timeout value.  Can one of you try it with the fix in https://github.com/Azure/go-amqp/pull/67 to see if it works?

It doesn't repro with an Azure servicebus broker.

**Comment by jhendrixMSFT** [comment]
I was able to set up a repro.  On Windows I get an "i/o timeout" instead of EOF though (I'm going to assume you're running on a non-Windows platform).  With the fix in #67 I can confirm the bug is fixed.

**Comment by jhendrixMSFT** [comment]
Fixed in v0.16.0
